### PR TITLE
[SR-3] base github CI

### PR DESCRIPTION
**Pull request by AKShipulya**
## Base GitHub CI implementation
* will merged after SR-1

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
```
Comment (N0TARGET) [inline .github/workflows/ci.yml]: let's use 21 java.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (N0TARGET) [inline .github/workflows/ci.yml:29]: is there only ubuntu runners?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```
Comment (N0TARGET) [inline .github/workflows/ci.yml]: I would do this using pattern against raised branch like 'story/SR*'

File: .github/workflows/ci.yml (hunk 2 above)
Comment (AKShipulya) [reply .github/workflows/ci.yml:29]: used Ubuntu because i know this system much better than any another one

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AKShipulya) [reply .github/workflows/ci.yml]: done

File: .github/workflows/ci.yml (hunk 3 above)
Comment (AKShipulya) [reply .github/workflows/ci.yml]: added patterns for `sr`, `main` and `release` branches. Note, now this made some contract for us - we must to use sr-prefix in our branches every time in 100% cases :smiley: !

**Pull request closed without merging**